packages/flutter_driver/lib/src/driver/vmservice_driver.dart

```diff
@@ -150,11 +149,31 @@ class VMServiceFlutterDriver extends FlutterDriver {
     }
 
     /// Waits for a signal from the VM service that the extension is registered.
-    /// Returns [_flutterExtensionMethodName]
-    Future<String> waitForServiceExtension() {
-      return isolate.onExtensionAdded.firstWhere((String extension) {
-        return extension == _flutterExtensionMethodName;
-      });
+    ///
+    /// Looks at the list of loaded extensions for the current [isolateRef], as
+    /// well as the stream of added extensions.
+    Future<void> waitForServiceExtension() async {
+      _log('Waiting for service extension');
```

Comment:
nit: What level of logging is this? Perhaps this should be for `--verbose` only? I imagine in normal circumstances a developer may not even know what "service extension" is. I think something like "Connecting to the application" might be more intuitive.